apps/ledger-live-desktop/src/newArch/features/DynamicContent/components/LogContentCardWrapper.tsx

```diff
@@ -20,24 +25,42 @@ const LogContentCardWrapper: React.FC<LogContentCardWrapperProps> = ({
 }) => {
   const ref = useRef<HTMLDivElement>(null);
   const isTrackedUser = useSelector(trackingEnabledSelector);
+  const anonymousUserNotifications = useSelector(anonymousUserNotificationsSelector);
+  const dispatch = useDispatch();
 
   const currentCard = useMemo(() => {
     const cards = braze.getCachedContentCards().cards;
     return cards.find(card => card.id === id);
   }, [id]);
 
   useEffect(() => {
-    if (!currentCard || !isTrackedUser) return;
+    if (!currentCard) return;
 
     const intersectionObserver = new IntersectionObserver(
       ([entry]) => {
         if (entry.intersectionRatio > PERCENTAGE_OF_CARD_VISIBLE) {
-          braze.logContentCardImpressions([currentCard]);
-          track("contentcard_impression", {
-            id: currentCard.id,
-            ...currentCard.extras,
-            ...additionalProps,
-          });
+          console.log("hereee");
+          if (isTrackedUser) {
+            braze.logContentCardImpressions([currentCard]);
+            track("contentcard_impression", {
+              id: currentCard.id,
+              ...currentCard.extras,
+              ...additionalProps,
+            });
+          } else if (
+            anonymousUserNotifications[currentCard.id as string] !==
+            currentCard?.expiresAt?.getTime()
+          ) {
+            // support new campaign or resumed campaign with the same id + different expiration date targeting anonymous users
+            setTimeout(() => {
```

Comment:
Why do we need to use a timeout, please?

Reply:
so , this timeout is used to wait a little bit before a ui change to mark the content card read (as if it's an api request) , Lucas told me that before the work on this feature it was 3 seconds. It's a time laps to avoid immediate disappear of the dot. Initially i just put 500 ms (like in the video) so it's can disappear in a reasonable time. But i should confirm with Anthony.

Reply:
simply it will take 3 seconds so that the seen status will be considered for the offline notifications.

Reply:
Actually it was 2 seconds my bad here is the code used before :
 
```
const { notificationsCards, logNotificationImpression, groupNotifications, onClickNotif } =
    useNotifications();

  const timeoutByUUID = useRef<Record<string, NodeJS.Timeout>>({});
  const handleInViewNotif = useCallback(
    (visible: boolean, uuid: keyof typeof timeoutByUUID.current) => {
      const timeouts = timeoutByUUID.current;

      if (notificationsCards.find(n => !n.viewed && n.id === uuid) && visible && !timeouts[uuid]) {
        timeouts[uuid] = setTimeout(() => {
          logNotificationImpression(uuid);
          delete timeouts[uuid];
        }, 2000);
      }
      if (!visible && timeouts[uuid]) {
        clearTimeout(timeouts[uuid]);
        delete timeouts[uuid];
      }
    },
    [logNotificationImpression, notificationsCards],
  );
```

Reply:
ok i'll change it ! so that we keep the same look & feel of a real api call , rather than suddenly/fast delete them

Reply:
✅ Done